core/src/consensus/tendermint/mod.rs

```diff
@@ -295,7 +298,10 @@ impl Tendermint {
     }
 
     fn to_step(&self, step: Step) {
-        self.extension.set_timer_step(step);
+        // Should be protected by step_change_lock
+        let counter = self.timeout_token_counter.load(AtomicOrdering::SeqCst);
+        self.extension.set_timer_step(step, counter);
```

Comment:
I think that we should pass the increased counter value.
The function's name is `set_timer_step`, so we need to pass the value to be set which is the increased counter.

Reply:
I think it's much better. I'll fix.